app/helpers/puppet_related_helper.rb

```diff
@@ -67,10 +67,36 @@ def interesting_puppetclasses(obj)
     classes.where(:id => class_vars)
   end
 
+  def extraction_warning
+    "being extracted to the foreman_puppet plugin and will be removed from core in 3.0"
```

Comment:
not crucial since i'm not sure how many translations we'll get in time for the extraction, but this string isn't marked for translation

Reply:
right, we would need to use 
```suggestion
    N_("being extracted to the foreman_puppet plugin and will be removed from core in 3.0")
```

and then

```ruby
 _("other part of the string %s") % extraction_warning
```

But there is not much we can do now unless it's crucial.

Reply:
https://github.com/theforeman/foreman/pull/8331